oeps/best-practices/oep-0067-bp-tools-and-technology.rst

```diff
@@ -0,0 +1,342 @@
+   configuration that provides an enforceable specification. The
+   `edX ESLint Config`_ is made available as an npm package that can be
+   installed into any Open edX package.
+
+   **Exception**: The `edX ESLint Config for ES5`_ may be used only where ES5
+   is in use. Both configs may be used in the same codebase through the
+   use of `ESLint configuration cascading`_.
+
+#. **JavaScript libraries should be installed via npm**
+
+   **Rationale**: It is important that JavaScript libraries are kept
+   up-to-date, and one key component is to make it as simple as possible
+   to perform upgrades. Projects that use npm maintain all of their
+   library requirements in a single `package.json` file, which can be
+   easily changed as the versions change.
+
+#. **JavaScript projects should publish a package.lock file**
+
+   **Rationale**: To keep dependencies up-to-date and builds consistent, JavaScript
+   projects should allow patch and minor upgrades in their package.json file and
+   commit a package-lock.json file. The package-lock.json file will keep a full list
+   of dependencies and their versions, ensuring when built for deployments the
+   version of libraries are consistent. This follows the same pattern used in edX
+   Python code - see `OEP 18`_ for more information. For more information on
+   package-lock files see `Package Lock`_.
+
+#. **Keep dependencies up to date by using Greenkeeper**
+
+   **Rationale**: JavaScript dependencies are updated constantly and can be difficult
+   to maintain over time. `Greenkeeper`_ makes this easier by automatically updating
+   dependencies in a package.json file and verifying the tests still work. When
+   possible, leverage Greenkeeper to ensure JavaScript software remains up to date.
+
+#. **JavaScript should be bundled using Webpack**
+
+   **Rationale**: `Webpack`_ is the tool of choice for code optimization and
+   bundling, as it is widely used throughout the industry, and because
+   it seamlessly handles modern code bases as well as all of the older
+   technologies used by edX, such as `AMD Modules`_. Webpack should be
+   implemented to handle as much of the "asset pipeline" as possible,
+   rather than passing this responsibility on to Django.
+
+#. **JavaScript dependencies should be managed with ES2015 Modules**
+
+   **Rationale**: JavaScript module systems allow front end code to specify
+   its dependencies and be grouped into bundles that minimize the assets
+   needed to provide page functionality. The most prevalent module syntax
+   is currently `ES2015 Modules`_, which should be adopted everywhere
+   edX code is written to the ES2015 spec or later.
+
+   **Exception**: Much of edX's existing (ES5) JavaScript uses the older
+   `AMD Modules`_ syntax for modules. AMD Modules are interoperable
+   with ES2015 Modules if Webpack is used for bundling, so AMD is an
+   acceptable module definition if the code must remain ES5.
+
+#. **CSS should be generated using Sass**
```

Comment:
```suggestion
#. **CSS should be generated using SCSS**
```

We want people to use the SCSS syntax and *not* the original Sass syntax, right? If so , let's put that on the first line.